src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -788,45 +798,59 @@ type internal ConsumerImpl internal (consumerConfig: ConsumerConfiguration, clie
 
         member this.AcknowledgeAsync (msgId: MessageId) =
             task {
-                connectionHandler.CheckIfActive()
+                let exn = connectionHandler.CheckIfActive()
+                if not (isNull exn) then 
+                    interceptors.OnAcknowledge(this, msgId, exn)
+                    raise exn
+                
                 mb.Post(Acknowledge(msgId, AckType.Individual))
             }
             
         member this.AcknowledgeAsync (msgs: Messages) =
             task {
-                connectionHandler.CheckIfActive()
+                let exn = connectionHandler.CheckIfActive()
+                if not (isNull exn) then 
+                    for msg in msgs do
+                        interceptors.OnAcknowledge(this, msg.MessageId, null)
```

Comment:
shouldn't be an exn there instead of null?

Reply:
Fixed